clients/src/main/java/org/apache/kafka/common/memory/GarbageCollectedMemoryPool.java

```diff
@@ -41,8 +41,8 @@
     private final Thread gcListenerThread;
     private volatile boolean alive = true;
 
-    public GarbageCollectedMemoryPool(long sizeBytes, int maxSingleAllocationSize, boolean strict, Sensor oomPeriodSensor) {
```

Comment:
adding an extra constructor and preserving the original signature (setting the sensor to null) would make rebasing on top of future kafka releases easier